OSX: AVCaptureDeviceTypeExternalUnknown & Invalid UTF-8

Having "issues" compling on macOS Sonoma 14.2.

...

```
[ 77%] Building C object modules/x11/CMakeFiles/x11.dir/x11.c.o
In file included from /Users/xxx/dev/foo/baresip3/BaresipBuild/baresip/modules/plc/plc.c:9:
In file included from /opt/homebrew/include/spandsp.h:75:
In file included from /opt/homebrew/include/spandsp/noise.h/Users/xxx/dev/foo/baresip3/BaresipBuild/baresip/modules/g722/g722.c::15:
In file included from /opt/homebrew/include/spandsp.h:7572:45: :
warning: /opt/homebrew/include/spandsp/noise.h:72:45:invalid UTF-8 in comment [-Winvalid-utf8]
 warning: invalid UTF-8 in comment [-Winvalid-utf8]The tolerance for each 1/3rd octave band is 
<A1><D3>3dB.
                                            ^
The tolerance for each 1/3rd octave band is <A1><D3>3dB.
                                            ^
[ 77%] Building C object modules/vp8/CMakeFiles/vp8.dir/sdp.c.o
In file included from 
/Users/xxx/dev/foo/baresip3/BaresipBuild/baresip/modules/g726/g726.c:11:
In file included from /opt/homebrew/include/spandsp.h:75:
/opt/homebrew/include/spandsp/noise.h:72:45: warning: invalid UTF-8 in comment [-Winvalid-utf8]
The tolerance for each 1/3rd octave band is <A1><D3>3dB.
                                            ^
[ 77%] Linking C shared module sdl.so

```
...

```
[ 99%] Built target baresip_exe

/Users/xxx/dev/foo/baresip3/BaresipBuild/baresip/modules/avcapture/avcapture.m:161:6: warning: 'AVCaptureDeviceTypeExternalUnknown' is deprecated: first deprecated in macOS 14.0 [-Wdeprecated-declarations]
                          AVCaptureDeviceTypeExternalUnknown]
                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                          AVCaptureDeviceTypeExternal
/Library/Developer/CommandLineTools/SDKs/MacOSX14.2.sdk/System/Library/Frameworks/AVFoundation.framework/Headers/AVCaptureDevice.h:563:38: note: 'AVCaptureDeviceTypeExternalUnknown' has been explicitly marked deprecated here
AVF_EXPORT AVCaptureDeviceType const AVCaptureDeviceTypeExternalUnknown API_DEPRECATED_WITH_REPLACEMENT("AVCaptureDeviceTypeExternal", macos(10.15, 14.0)) API_UNAVAILABLE(ios, macCatalyst, tvos, visionos) API_UNAVAILABLE(watchos);
                                     ^
/Users/xxx/dev/foo/baresip3/BaresipBuild/baresip/modules/avcapture/avcapture.m:407:6: warning: 'AVCaptureDeviceTypeExternalUnknown' is deprecated: first deprecated in macOS 14.0 [-Wdeprecated-declarations]
                          AVCaptureDeviceTypeExternalUnknown]
                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                          AVCaptureDeviceTypeExternal
/Library/Developer/CommandLineTools/SDKs/MacOSX14.2.sdk/System/Library/Frameworks/AVFoundation.framework/Headers/AVCaptureDevice.h:563:38: note: 'AVCaptureDeviceTypeExternalUnknown' has been explicitly marked deprecated here
AVF_EXPORT AVCaptureDeviceType const AVCaptureDeviceTypeExternalUnknown API_DEPRECATED_WITH_REPLACEMENT("AVCaptureDeviceTypeExternal", macos(10.15, 14.0)) API_UNAVAILABLE(ios, macCatalyst, tvos, visionos) API_UNAVAILABLE(watchos);
                                     ^
2 warnings generated.

```

and when starting baresip:
...
```
vidcodec: VP8
vidcodec: VP9
vidsrc: avcapture
2023-12-18 14:48:56.268 baresip[71447:2092711] WARNING: AVCaptureDeviceTypeExternal is deprecated for Continuity Cameras. Please use AVCaptureDeviceTypeContinuityCamera and add NSCameraUseContinuityCameraDeviceType to your Info.plist.
medianat: stun
medianat: turn
medianat: ice

```
...

--- a/modules/avcapture/avcapture.m
+++ b/modules/avcapture/avcapture.m
@@ -158,7 +158,12 @@ static void vidframe_set_pixbuf(struct vidframe *f, const CVImageBufferRef b)
 		AVCaptureDeviceDiscoverySession
 		discoverySessionWithDeviceTypes:
 			@[AVCaptureDeviceTypeBuiltInWideAngleCamera,
+#if (__IPHONE_OS_VERSION_MIN_REQUIRED >= 170000 || \
+     __MAC_OS_X_VERSION_MIN_REQUIRED >= 140000)
+			  AVCaptureDeviceTypeExternal]
+#else
 			  AVCaptureDeviceTypeExternalUnknown]
+#endif
 		mediaType:AVMediaTypeVideo
 		position:pos
 	];
@@ -404,7 +409,12 @@ static int module_init(void)
 		AVCaptureDeviceDiscoverySession
 		discoverySessionWithDeviceTypes:
 			@[AVCaptureDeviceTypeBuiltInWideAngleCamera,
+#if (__IPHONE_OS_VERSION_MIN_REQUIRED >= 170000 || \
+     __MAC_OS_X_VERSION_MIN_REQUIRED >= 140000)
+			  AVCaptureDeviceTypeExternal]
+#else
 			  AVCaptureDeviceTypeExternalUnknown]
+#endif
 		mediaType:AVMediaTypeVideo
 		position:AVCaptureDevicePositionUnspecified
 	];

